rasa/importers/autoconfig.py

```diff
@@ -0,0 +1,241 @@
+
+    autoconfig_lines = _get_commented_out_autoconfig_lines(config, auto_configured_keys)
+
+    with open(config_file_path, "r+", encoding=io_utils.DEFAULT_ENCODING) as f:
+        lines = f.readlines()
+        updated_lines = _get_lines_including_autoconfig(lines, autoconfig_lines)
+        f.seek(0)
+        for line in updated_lines:
+            f.write(line)
+
+    if auto_configured_keys:
+        auto_configured_keys = common_utils.transform_collection_to_sentence(
+            auto_configured_keys
+        )
+        cli_utils.print_info(
+            f"The configuration for {auto_configured_keys} was chosen automatically. It "
+            f"was written into the config file at '{config_file_path}'."
+        )
+
+
+def _is_config_file_as_expected(
+    config_file_path: Text, missing_keys: List[Text], auto_configured_keys: List[Text],
+) -> bool:
+    try:
+        content = io_utils.read_config_file(config_file_path)
+    except ValueError:
+        content = ""
+
+    return (
+        bool(content)
+        and missing_keys == _get_missing_config_keys(content)
+        and auto_configured_keys == _get_unspecified_autoconfigurable_keys(content)
+    )
+
+
+def _add_missing_config_keys_to_file(
+    config_file_path: Text, missing_keys: List
+) -> None:
+    if missing_keys:
+        with open(config_file_path, "a", encoding=io_utils.DEFAULT_ENCODING) as f:
+            for key in missing_keys:
+                f.write(f"{key}:\n")
+
+
+def _get_lines_including_autoconfig(
+    lines: List[Text], autoconfig_lines: Dict[Text, List[Text]],
+) -> List[Text]:
+    auto_configured_keys = autoconfig_lines.keys()
+
+    lines_with_autoconfig = []
+    remove_comments_until_next_uncommented_line = False
+    for line in lines:
+        insert_section = None
+
+        # remove old auto configuration
+        if remove_comments_until_next_uncommented_line:
+            if line.startswith("#"):
+                continue
+            remove_comments_until_next_uncommented_line = False
+
+        # add an explanatory comment to auto configured sections
+        for key in auto_configured_keys:
+            if line.startswith(f"{key}:"):  # start of next auto-section
+                line = line + COMMENTS_FOR_KEYS[key]
+                insert_section = key
+                remove_comments_until_next_uncommented_line = True
+
+        lines_with_autoconfig.append(line)
+
+        if not insert_section:
+            continue
+
+        # add the auto configuration (commented out)
+        lines_with_autoconfig += autoconfig_lines[insert_section]
+
+    return lines_with_autoconfig
+
+
+def _get_commented_out_autoconfig_lines(
+    config: Dict[Text, Any], autoconfigured: List[Text]
+) -> Dict[Text, List[Text]]:
+    import ruamel.yaml as yaml
+
+    yaml_parser = yaml.YAML()
+    yaml_parser.indent(mapping=2, sequence=4, offset=2)
+
+    autoconfig_lines = {}
+
+    for key in autoconfigured:
```

Comment:
```suggestion
    for key, value in autoconfigured.items():
```

Reply:
?? `autoconfigured` is just a `Set` with all the auto configured keys. I'll rename it for more clarity.

Reply:
oh dear 🙈 It's fine 🙈